src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,78 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static List<User> users;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        users = new ArrayList<>();
+        users.add(new User("UserX2023", "pass6789", 35));
+        users.add(new User("JohnDoe23", "john4567", 28));
+        users.add(new User("Anna1998", "password123", 25));
+    }
+
+    @Test
+    void userNotInDb_isOK() {
+        User actual = registrationService.register(new User("Ivan2004", "1234567", 20));
+        assertEquals(new User("Ivan2004", "1234567", 20), actual);
+    }
+
+    @Test
+    void userAge_isNotOK() {
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(new User("Ivan2004", "1234567", 10));
+        });
+    }
+
+    @Test
+    void userLogin_isNotOK() {
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(new User("Ivan", "1234567", 18));
```

Comment:
The test name `userLogin_isNotOK` is not descriptive enough. It should be clear what about the user login is not OK. If it's testing for a null login, name it `register_nullLogin_exceptionThrown`.